superset-frontend/src/dashboard/actions/dashboardState.js

```diff
@@ -474,6 +480,21 @@ export function addSliceToDashboard(id, component) {
       if (selectedSlice && selectedSlice.viz_type === 'filter_box') {
         dispatch(addFilter(id, component, selectedSlice.form_data));
       }
+
+      setTimeout(() => {
```

Comment:
You told me about this timeout but the more I think about it the more I am concerned that it might fail in a few instances, especially in large dashboards. We should probably look into how the dashboard should signal that all charts have finished querying. I think this is important for other use cases as well. @rusackas what do you think?

Reply:
Agreed. I'm not sure what the 500ms is based on, but it feels arbitrary compared to what's likely to happen in reality. I'm hopeful that there's something we can `await` here. I'm not sure exactly what yet... will need to dig a bit deeper.

Reply:
Yes. In this case it is not only necessary to wait for the chart data from the server, but the exact timing is after the chart transform props, after the colon called. So I'm using `postTransformProps` to handle this (an api that seems to be rarely used, please let me know if this makes sense). See the video, I simulated a weak network environment, but the colors still updated after adding the chart.

https://user-images.githubusercontent.com/11830681/154857051-d76764cb-95cf-4bfd-b2dc-233bc95d9db3.mov